src/v2/Apps/Settings/Routes/DeleteAccount/DeleteAccountRoute.tsx

```diff
@@ -0,0 +1,83 @@
+import { Button, Checkbox, Separator, Text, TextArea, useToasts
+} from "@artsy/palette"
+import * as React from "react"
+import { useState } from "react"
+import { RouterLink } from "v2/System/Router/RouterLink"
+import { deleteMyAccount } from "../../Mutations/DeleteMyAccount"
+import { useSystemContext } from "v2/System/SystemContext"
+
+export const DeleteAccountRoute = () => {
+  const { relayEnvironment } = useSystemContext()
+
+  const [isConfirmed, setConfirmed] = useState(false)
+  const [enteredReason, setEnteredReason] = useState("")
+  const isDisabled = !isConfirmed || !enteredReason
+  const { sendToast } = useToasts()
+
+  const handleDeleteMyAccount = async () => {
```

Comment:
We want some UI feedback here so starting the spinner loading state on the button here.